src/main/java/org/folio/coursereserves/util/CRUtil.java

```diff
@@ -299,211 +263,141 @@ public static void populateReserveCopiedItemFromJson(Reserve reserve, JsonObject
 
   public static Future<JsonObject> lookupItemHoldingsInstanceByItemId(String itemId,
       Map<String, String> okapiHeaders, Context context) {
-    Promise promise = Promise.promise();
-    JsonObject result = new JsonObject();
     logger.info("Making request for item at {}/{}", ITEMS_ENDPOINT, itemId);
-    makeOkapiRequest(context.owner(), okapiHeaders, ITEMS_ENDPOINT + "/" + itemId,
-        HttpMethod.GET, null, null, 200).onComplete(itemRes -> {
-      if(itemRes.failed()) {
-        logger.error("Unable to lookup item by id {}", itemId);
-        promise.fail(itemRes.cause());
-      } else {
-        JsonObject itemJson = itemRes.result();
-        String holdingsId = itemJson.getString("holdingsRecordId");
-        result.put("item", itemJson);
-        logger.info("Making request for holdings at {}/{}", HOLDINGS_ENDPOINT, holdingsId);
-        makeOkapiRequest(context.owner(), okapiHeaders, HOLDINGS_ENDPOINT + "/" + holdingsId,
-            HttpMethod.GET, null, null, 200).onComplete(holdingsRes -> {
-          if(holdingsRes.failed()) {
-            promise.fail(holdingsRes.cause());
-          } else {
-            JsonObject holdingsJson = holdingsRes.result();
-            String instanceId = holdingsJson.getString("instanceId");
-            result.put("holdings", holdingsJson);
-            logger.info("Making request for instance at {}/{}", INSTANCES_ENDPOINT, instanceId);
-            makeOkapiRequest(context.owner(), okapiHeaders, INSTANCES_ENDPOINT
-                + "/" + instanceId, HttpMethod.GET, null, null, 200).onComplete(
-                instanceRes -> {
-              if(instanceRes.failed()) {
-                promise.fail(instanceRes.cause());
-              } else {
-                JsonObject instanceJson = instanceRes.result();
-                result.put("instance", instanceJson);
-                logger.info("Inventory lookup complete");
-                promise.complete(result);
-              }
-            });
-          }
+    return makeOkapiRequest(context.owner(), okapiHeaders, ITEMS_ENDPOINT + "/" + itemId,
+        HttpMethod.GET, null, null, 200).compose(itemJson -> {
```

Comment:
For me it's more readable if there's a line break before each `.compose` and `.map`.